### DISCUSSION TITLE
Compilation time

### DISCUSSION THREAD

**Discussion by daniele-baccega** [FLAMEGPU/FLAMEGPU2#1115] [Q&A] [answered]
Hi,
is it normal that it takes about 6-7 minutes every time to compile the main.cu? I do not have too much code..

Thank you!

**Comment by Robadob** [comment, marked as answer]
No.

The only very slow compilation that we're aware of is when RTC agent function are compiled (at runtime) which make use of `FLAMEGPU_ENABLE_GLM`. We've done some work to improve this, but the main limitation hangs with a dependency that processes files for runtime compilation (https://github.com/FLAMEGPU/FLAMEGPU2/issues/602).

If your model is large you can split it across multiple compilation units, and take advantage of parallel compilation (Visual studio will do this automatically, however I believer `gcc`/`make` requires you to pass `-j<threads>` or if using `cmake --build .` instead `--parallel <threads>`). I don't think we don't have a guide for splitting models across compilation units, however a [model](https://github.com/primagesheffield/flamegpu2-neuroblastoma/tree/orchestrator_release/src) that I previously worked with is open-source and is split across many files.

Likewise, ensure you are setting `CMAKE_CUDA_ARCHITECTURES` to a single (and the correct) SM ver for your targeted GPU, otherwise all device code will be getting compiled multiple times (once per major architecture). You can find the full list of GPU->arch [here](https://developer.nvidia.com/cuda-gpus), e.g. `7.0` would be entered as `70` within CMake.
